### PR TITLE
[ENH] Network Of Groups: Add a new widget

### PR DESCRIPTION
**Pull request by VesnaT**
##### Issue

Reduce by value: Widget to create a network with nodes corresponding to values of some categorical feature (Department), and two values are connected, if there exists a pair of nodes in the original network that are connected and have this values. Its weight is the sum of weights.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: orangecontrib/network/widgets/OWNxGroups.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    def _map_into_feature_values(self, row, col):
+        n_edges = self.network.number_of_edges()
+        selected_column = self.effective_data.get_column_view(self.feature)[0]
+        mapped_row = np.empty(n_edges)
+        mapped_col = np.empty(n_edges)
+        for node in range(self.network.number_of_nodes()):
+            value = selected_column[node]
+            mapped_row[row == node] = value
+            mapped_col[col == node] = value
+        return mapped_row, mapped_col
```
Comment (janezd) [inline orangecontrib/network/widgets/OWNxGroups.py]: Is this the same as

```
    def _map_into_feature_values(self, row, col):
        selected_column = self.effective_data.get_column_view(self.feature)[0]
        return selected_column[row], selected_column[col]
```

File: orangecontrib/network/widgets/OWNxGroups.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        self.network = network
+        self.__set_input_label_text()
+
+    @Inputs.data
+    def set_data(self, data):
+        self.data = data
+
+    def handleNewSignals(self):
+        if self.feature_model:
+            self.closeContext()
```
Comment (janezd) [inline orangecontrib/network/widgets/OWNxGroups.py]: This `if` is unnecessary; `closeContext` contains a check whether a context is currently open.

File: orangecontrib/network/widgets/OWNxGroups.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        # find unique edges
+        mask = row > col
+        row[mask], col[mask] = col[mask], row[mask]
+
+        array = np.vstack((row, col))
+        (row, col), inverse = np.unique(array, axis=1, return_inverse=True)
+
+        # assign each edge the sum of weights of belonging original edges
+        weights = np.array(
+            [np.sum(weights[inverse == i]) for i in range(len(row))])
```
Comment (janezd) [inline orangecontrib/network/widgets/OWNxGroups.py]: This is cool. And I learned something new about `np.unique`.

File: orangecontrib/network/widgets/OWNxGroups.py (hunk 1 above)
Comment (VesnaT) [reply orangecontrib/network/widgets/OWNxGroups.py]: It is the same ass :+1:
```
def _map_into_feature_values(self, row, col):
        selected_column = self.effective_data.get_column_view(self.feature)[0]
        return (selected_column[row].astype(np.float64),
                selected_column[col].astype(np.float64))
```

**Comment by VesnaT** [comment]
Maybe we can agree on the input and output Info and I could make those consistent on all widget. As well as the signal names.

**Pull request merged**